src/omotes_simulator_core/solver/utils/fluid_properties.py

```diff
@@ -13,13 +13,76 @@
 #  You should have received a copy of the GNU General Public License
 #  along with this program.  If not, see <https://www.gnu.org/licenses/>.
 """This module contains the fluid properties class."""
-import csv
-from pathlib import Path
 
-from scipy.interpolate import make_interp_spline
+import numpy as np
+from fluidprop import FluidProperties
 
 
-class FluidProperties:
+class Interpolation:
+    """Class to enable interpolation in a set of data points."""
+
+    def __init__(self, x: list[float], y: list[float], order: int = 5):
+        """Constructor of the interpolation class.
+
+        It stores the x and y data and determine the coefficients of a polynomial of the given
+        order. This polynomial is used to interpolate the data. Please note that no checks on
+        the bounds have been included.
+
+        :param x: The x values of the data points.
+        :param y: The y values of the data points.
+        :param order: The order of the polynomial to use for the interpolation.
+        """
+        self.x = x
+        self.y = y
+        self.order = order
+        self.coefficients = np.polyfit(x, y, order)
+        self._check_interpolation()
+
+    def _check_interpolation(self) -> None:
+        """Check if the interpolation gives acceptable results.
+
+        For every data point the difference with the interpolated value is calculated.
+        If this difference is more than 2% and value error is raised.
+        """
+        for i in range(len(self.x)):
+            if self.y[i] == 0:
+                continue
+            error = abs(self.y[i] - self(self.x[i])) / self.y[i]
+            if error == np.inf:
+                continue
+            if error > 0.02:
+                raise ValueError("Interpolation error: error is more then 2%.")
+
+    def __call__(self, value: float) -> float:
+        """Returns the interpolated value at a given point.
+
+        The calculate the value at the given point, the coefficients of the polynomial are used.
+        A backwards loop is used, since the first value in the list is the one which is multiplied
+        with the highest order. In this way we can step by step multiply the temp value with the
+        value at the given point.
+
+        :param value: The value to interpolate.
+        :return: The interpolated value at the given point.
+        """
+        self._check_bounds(value=value)
+        result = 0
+        temp_value = 1.0
+        for i in range(self.order, -1, -1):
```

Comment:
because:
This forms part of the old polynomial API. Since version 1.4, the new polynomial API defined in [numpy.polynomial](https://numpy.org/doc/2.0/reference/routines.polynomials-package.html#module-numpy.polynomial) is preferred. A summary of the differences can be found in the [transition guide](https://numpy.org/doc/2.0/reference/routines.polynomials.html).
I will look into using the polynomial class

Reply:
I checked, but using the polynomial class is slower. 
In a simple test case even a factor 2